src/common/meta/src/kv_backend/rds/postgres.rs

```diff
@@ -193,40 +193,42 @@ fn pg_generate_in_placeholders(from: usize, to: usize) -> Vec<String> {
 /// Factory for building sql templates.
 struct PgSqlTemplateFactory<'a> {
     table_name: &'a str,
+    schema: Option<&'a str>,
 }
 
 impl<'a> PgSqlTemplateFactory<'a> {
-    /// Creates a new [`SqlTemplateFactory`] with the given table name.
-    fn new(table_name: &'a str) -> Self {
-        Self { table_name }
+    /// Creates a new factory with optional schema.
+    fn with_schema(table_name: &'a str, schema: Option<&'a str>) -> Self {
+        Self { table_name, schema }
     }
 
     /// Builds the template set for the given table name.
     fn build(&self) -> PgSqlTemplateSet {
-        let table_name = self.table_name;
+        let table_ident = self.table_name;
+        let table_ref = match self.schema {
+            Some(schema) if !schema.is_empty() => format!("\"{}\".\"{}\"", schema, table_ident),
+            _ => format!("\"{}\"", table_ident),
+        };
```

Comment:
Prefer introducing a function here, along with a unit test to validate its behavior.